### ISSUE TITLE
permission question

### ISSUE THREAD

**Issue by jumbohei** [checkpoint-restore/criu#1901]
Hi, I have a question regarding to the permission and much appreciated for inputs. Let’s say if a user A who writes a program and runs as a user and then he/she wants to take a snapshot and I don’t want to give this person the root access. Is that doable? If yes, can user A goes into the image and modify the UID to 0, which is as root, to restore and run that application as root? Thanks!

**Comment by rst0git** [comment]
@jumbohei It might be worth looking at [user namespaces](https://www.man7.org/linux/man-pages/man7/user_namespaces.7.html). You can set `UID=0` as follows:

```
$ id -u
1000
$ unshare -Umr bash
# id -u
0
```

**Comment by rst0git** [comment]
> I don’t want to give this person the root access. Is that doable?

Yes, checkpoint/restore can be used with non-root users as well: https://github.com/checkpoint-restore/criu/pull/1155
